posts/2021/12/conda-recipe-grayskull.md

```diff
@@ -0,0 +1,124 @@
+
+To publish your package, you need to write its recipe and then submit it through a pull request on the [staged-recipes](https://github.com/conda-forge/staged-recipes) repository of conda-forge. This pull request will be community reviewed and if approved, your package will become available on the conda-forge channel.
+
+<p align="center">
+    <img
+     alt="Conda-build transforms recipes into packages"
+     src="/images/2021/12/conda-build.png">
+    <i><br>Conda-build transforms recipes into packages</i>
+</p>
+
+## Grayskull — the automatic Conda recipe generator
+The process to publish a package on conda-forge is simple and straight forward: write the recipe, create a pull request, wait for the community review. But writing recipes is not simple. It can be error prone and tiresome.
+To alleviate this conda-forge provides a template recipe that can be used as a starting point and edited according to one’s needs. But even that could be too intimidating for someone new to packaging and recipes.
+
+Grayskull solves this problem. Grayskull is an automatic conda recipe generator, with a focus on conda-forge. It generates concise and accurate recipes very quickly, provided the package is available on PyPI.
+All you have to do is pass in the name of the Python package to Grayskull and it will generate its recipe for you.
+
+<p align="center">
+    <img
+     alt="Grayskull automates recipe generation"
+     src="/images/2021/12/package_name.png">
+    <i><br>Grayskull automates recipe generation</i>
+</p>
+
+Now that you have the package recipe, you create a pull request on the staged-recipes repository and wait for someone from the conda-forge community to review it. You know the drill!
+
+## But what if a package is not published on PyPI?
+Yes. Unfortunately, that’s where Grayskull falls a little short. It only generates recipes for Python packages available on [PyPI](https://pypi.org/). This prerequisite leaves out a number of Python packages otherwise available online.
+
+## My life’s purpose — making Grayskull more versatile
+During my internship at Quansight Labs, I added the ability to generate recipes from GitHub repositories.
+This way, a package that has not been published on PyPI but lives as a GitHub repository may have its recipe automatically generated with Grayskull.
+
+Before my work, Grayskull would extract metadata of the package from two sources: PyPI and the source distribution (often abbreviated as ‘sdist’). It then merged the PyPI metadata and the sdist metadata and used the resulting information to generate the final recipe.
+For Grayskull to accept packages coming from Github, I had to bypass some parts of that logic and patch others.
+
+For a package not published on PyPI, the PyPI metadata doesn’t exist. So for a GitHub package, I made Grayskull skip the part where it extracts metadata from PyPI. This way, only the sdist metadata was used to generate the recipe.
```

Comment:
There are a number of inconsistent capitalizations for github, Github, GitHub (the latter is the correct one). Might be worthwhile checking capitalization for other things as well, such as conda-forge, conda-build, etc.

Reply:
Thank you for pointing out, @melissawm ! :)
Done!